pkg/controller/atlasproject/auditing.go

```diff
@@ -59,12 +81,24 @@ func auditingInSync(atlas *mongodbatlas.Auditing, spec *v1.Auditing) bool {
 		return true
 	}
 
-	if isAuditingEmpty(atlas) || isAuditingEmpty(spec) {
-		return false
+	specAsAtlas := &mongodbatlas.Auditing{
+		AuditAuthorizationSuccess: toptr.MakePtr(false),
+		Enabled:                   toptr.MakePtr(false),
+	}
+
+	if !isAuditingEmpty(spec) {
+		specAsAtlas = spec.ToAtlas()
+	}
+
+	if isAuditingEmpty(atlas) {
+		atlas = &mongodbatlas.Auditing{
+			AuditAuthorizationSuccess: toptr.MakePtr(false),
+			Enabled:                   toptr.MakePtr(false),
+		}
 	}
```

Comment:
I do not understand this change, can you elaborate?

Reply:
While unlikely to happen, this is a protection once the response from the client is a pointer. Given that, if for any reason it returns no error and nil as a response, we assume default values.